processor/transformer/transformer.go

```diff
@@ -266,12 +252,7 @@ func NewTransformer(conf *config.Config, log logger.Logger, stat stats.Stats, op
 
 // Transform function is used to invoke destination transformer API
 func (trans *handle) Transform(ctx context.Context, clientEvents []TransformerEvent, batchSize int) Response {
-	if len(clientEvents) == 0 {
-		return Response{}
-	}
-	destType := clientEvents[0].Destination.DestinationDefinition.Name
-	transformURL := trans.destTransformURL(destType)
-	return trans.transform(ctx, clientEvents, transformURL, batchSize, destTransformerStage)
+	return trans.transform(ctx, clientEvents, trans.destTransformURL(clientEvents[0].Destination.DestinationDefinition.Name), batchSize, destTransformerStage)
```

Comment:
Shouldn't it be safe to check the length? even if the current caller ensures a empty slice is not sent?

Reply:
We have a check in the caller [here](https://github.com/rudderlabs/rudder-server/blob/master/processor/processor.go#L2662)

We can be safe but we should be safe even in `UserTransform` and `Validate` too as they are similar. Felt removing is the best way forward

Reply:
Just looking at this function: It is just doing a pass through. Couldn't we just get rid of this and rename trans.transform to trans.Transform?

Reply:
@sivashanmukh will take it up in a new PR. Will merge this PR for now